sentry/src/middleware/channel.rs

```diff
@@ -0,0 +1,43 @@
+use crate::db::field::{BigNumPg, ChannelIdPg};
+use crate::db::DbPool;
+use bb8_postgres::tokio_postgres::{types::Json, Row};
+use hyper::{Body, Request};
+use primitives::channel::ChannelId;
+use primitives::{Channel, ChannelSpec};
+use std::error;
+
+pub async fn channel_load(
+    mut req: Request<Body>,
+    (pool, id): (DbPool, &ChannelId),
+) -> Result<Request<Body>, Box<dyn error::Error>> {
+    let channel_id = hex::encode(id);
+
+    let channel = pool
+        .run(move |connection| {
+            async move {
+                match connection.prepare("SELECT channel_id, creator, deposit_asset, deposit_amount, valid_until, spec FROM channels WHERE channel_id = $1").await {
+                    Ok(select) => match connection.query_one(&select, &[&channel_id]).await {
+                            Ok(row) => Ok((channel_map(&row), connection)),
+                            Err(e) => Err((e, connection)),
+                        },
+                    Err(e) => Err((e, connection)),
+                }
+            }
+        })
+        .await?;
+
+    req.extensions_mut().insert(channel);
+
+    Ok(req)
+}
+
+fn channel_map(row: &Row) -> Channel {
```

Comment:
Probably that should go out with the fetching of channel(s). See comment above.

Reply:
It's just easier to show you the error:

```
error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> sentry/src/middleware/channel.rs:34:1
   |
34 | impl From<Row> for Channel {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^ impl doesn't use types inside crate
   |
   = note: the impl does not reference only types defined in this crate
   = note: define and implement a trait or new type instead

error: aborting due to previous error
```

If we keep the `field` module inside `sentry`, this cannot be done, since you cannot impl traits from one crate for types in another.

Reply:
> Also, `get` can panic should we prefer `get_opt`?
Good point, there is also `try_get`, but they panic in the cases of:
`Panics if the index is out of bounds or if the value cannot be converted to the specified type.`
Which means Dev mistake, which means it should panic in that case. We should handle the error.

Reply:
The From<From> is for ChannelPg